File: src/Main/lua/ga/corelauncher/Classes/WindowControl.lua
local WindowControl = Import("ga.corelauncher.Libraries.ClassCreator")(
    "WindowControl",
    function (self)
        
    end,
    {
        Close = function(self)
            CoreLauncher.BrowserWindow:close()
        end,

        Minimize = function(self)
            CoreLauncher.BrowserWindow:minimize()
        end,

        Maximize = function(self)
            CoreLauncher.BrowserWindow:maximize()
        end,

        Restore = function(self)
            CoreLauncher.BrowserWindow:restore()
        end,
        
        OpenExternal = function(self, Url)
            CoreLauncher.Electron.shell:openExternal(Url)
        end
    }
)

return WindowControl

File: src/Main/js/ga/corelauncher/Classes/WindowControl.js
class WindowControl {
    Close() {
        CoreLauncher.BrowserWindow.close()
    }
    
    Minimize() {
        CoreLauncher.BrowserWindow.minimize()
    }
    
    Maximize() {
        CoreLauncher.BrowserWindow.maximize()
    }
    
    Restore() {
        CoreLauncher.BrowserWindow.restore()
    }
    
    OpenExternal(Url) {
        CoreLauncher.Electron.shell.openExternal(Url)
    }
}

module.exports = WindowControl